generator/templates/enums/template.h

```diff
@@ -0,0 +1,22 @@
+{#- String based enum -#}
+{% include 'copyright.txt' %}
+{% block imports -%}
+{%- for import in imports %}
+#import "{{import}}.h"
+{%- endfor %}
+{%- endblock -%}
+{%- block body %}
+{% include 'description.jinja' %}
+typedef SDLEnum {{ name }} SDL_SWIFT_ENUM{{ending}};
+{% for param in params %}
+{%- include 'description_param.jinja' %}
+{% if deprecated or param.deprecated -%}
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+{%- endif %}
+extern {{ name }} const {{ name }}{{param.name}}{{ " __deprecated" if param.deprecated and param.deprecated }};
```

Comment:
Enums don't need to modify their object based on the reserved keywords unless the entire value (remember we prefix with SDL and the enum type) matches the reserved keyword.

Reply:
so if keyword would be `SDLDisplayTypeCID` then it should be renamed as follow?
```objc
//DisplayType.h
extern SDLDisplayType const SDLDisplayTypeCidParam;
//DisplayType.m
SDLDisplayType const SDLDisplayTypeCidParam = @"CID_PARAM";
```

Reply:
If the keyword is CID, it would be:

```objc
//DisplayType.h
extern SDLDisplayType const SDLDisplayTypeCid;
//DisplayType.m
SDLDisplayType const SDLDisplayTypeCidParam = @"CID";
```

If the keyword were `SDLDisplayTypeCID` it would be:

```objc
//DisplayType.h
extern SDLDisplayType const SDLDisplayTypeCidParam;
//DisplayType.m
SDLDisplayType const SDLDisplayTypeCidParam = @"CID";
```

The string value *MUST* remain as the constant (and having a string be a keyword is fine in any language anyway), or else SDL will fail.

Reply:
`@"{{param.origin}}"` will never be changed.
`{{param.name}}` could be changed if much with with following logic:
```python3
if any(re.search(r'^(sdl)?({})?{}$'.format(item_name.casefold(), name.casefold()), k) for k in self.key_words):
```
done in https://github.com/smartdevicelink/sdl_ios/pull/1556/commits/8d538469711fe0690157629c5f8d0c361b29cdff

Reply:
This doesn't seem to be completely fixed. `SDLImageType`'s `static` value: `SDLImageTypeStatic` is being turned into `SDLImageTypeStaticParam`. This shouldn't happen because `SDLImageTypeStatic` is not a keyword even though `static` is.

Reply:
Same for `SDLAppHMIDefaultParam`, that should be `SDLAppHMIDefault`.

Reply:
Same for `SDLResultSuccessParam`. It should be `SDLResultSuccess`.

Reply:
changed to following regex:
```regex
r'^(sdl)?({}){}$'.format(item_name.casefold(), name.casefold())
```